File: impact_lab/backend/app/schemas/capacity.py
from pydantic import BaseModel, Field
from typing import List, Optional
from app.schemas.priorizacion import CamelModel

class BoxCapacity(CamelModel):
    id: str
    cesfam_name: str
    box_number: int
    box_type: str  # "MEDICO" | "ENFERMERA" | "MULTIDISCIPLINARIO"
    is_active: bool
    weekly_hours_capacity: int
    current_weekly_occupancy: int

class ReferralQuota(CamelModel):
    id: str
    cesfam_name: str
    hospital_target: str
    specialty: str  # "DIABETOLOGIA" | "NEFROLOGIA" | "PIE_DIABETICO"
    monthly_quota_limit: int
    monthly_quota_used: int
    valid_month: str

class CapacitySummaryResponse(CamelModel):
    total_boxes: int
    active_boxes: int
    total_weekly_hours: int
    occupied_weekly_hours: int
    box_utilization_rate: float
    referral_quotas: List[ReferralQuota]

class CreateReferralPayload(CamelModel):
    patient_id: str
    target_facility: str
    target_specialty: str
    clinical_reason: str
    referring_physician: str

File: impact_lab/src/types/capacity.ts
export type UserRole = 'MEDICO_CONTRALOR' | 'ENFERMERA_GESTORA' | 'ADMIN_SOMO' | 'AUDITOR_MINSAL';

export interface BoxCapacity {
  id: string;
  cesfamName: string;
  boxNumber: number;
  boxType: 'MEDICO' | 'ENFERMERA' | 'MULTIDISCIPLINARIO';
  isActive: boolean;
  weeklyHoursCapacity: number;
  currentWeeklyOccupancy: number;
}

export interface ReferralQuota {
  id: string;
  cesfamName: string;
  hospitalTarget: string;
  specialty: 'DIABETOLOGIA' | 'NEFROLOGIA' | 'PIE_DIABETICO';
  monthlyQuotaLimit: number;
  monthlyQuotaUsed: number;
  validMonth: string;
}

export interface CapacitySummary {
  totalBoxes: number;
  activeBoxes: number;
  totalWeeklyHours: number;
  occupiedWeeklyHours: number;
  boxUtilizationRate: number;
  referralQuotas: ReferralQuota[];
  boxes: BoxCapacity[];
}
